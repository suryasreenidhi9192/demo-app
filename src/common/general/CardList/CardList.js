import React from 'react'
import PropTypes from 'prop-types';

import './CardList.scss'

/**
 * @public
 * Simple list generating component accepting an Array of objects with header and body keys
 * @param  {Array}      props.listData   List of "card" objects to map, generating "cards" with "header" and "body" keys
 * @return {Node}                        React Component
*/
const CardList = ({ listData }) => {

    return (

        <ul className="card-list">

            { listData.map((o, i) => (

                <li className="card" key={i}>

                    <div className="header">
                        { o.header }
                    </div>

                    <div className="body">
                        { o.body }
                    </div>

                </li>

            )

            )}

            <li className="card empty-card" />
            <li className="card empty-card" />
            <li className="card empty-card" />

        </ul>

    )

}

CardList.propTypes = {
    /** List of "card" objects to map, generating "cards" */
    listData: PropTypes.arrayOf(
        PropTypes.shape({
            /** React Component or string to render at the top of the Alert */
            header: PropTypes.node,
            /** Content to present as the body of the "card" */
            body: PropTypes.node
        })
    )
}

export default CardList
