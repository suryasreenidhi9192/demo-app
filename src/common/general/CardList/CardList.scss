@import "../../shared/css/variables";

.card-list {

    display: flex; flex-wrap: wrap;
    margin-left: -16px; margin-top: -16px;

    .card {

        flex: 1 1 auto;
        width: 300px;
        margin-left: 16px; margin-top: 16px;
        border: var(--border);

        &.empty-card {
            border-color: transparent;
        }

        .header {
            font-size: 18px; color: var(--text-muted);
            padding:16px;
            border-bottom: var(--border);
            color: var(--text);
        }

        .body { padding:16px; font-size:14px; }

    }

}
