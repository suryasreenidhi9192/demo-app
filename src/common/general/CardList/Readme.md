### CardList
```jsx
    <CardList listData={[
        {
            header: 'Card 1 Title',
            body: 'Body 1'
        },{
            header: 'Card 2 Title',
            body: 'Body 2'
        },{
            header: 'Card 3 Title',
            body: 'Body 3'
        },{
            header: 'Card 4 Title',
            body: 'Body 4'
        },{
            header: 'Card 5 Title',
            body: 'Body 5'
        },
    ]}/>
```
