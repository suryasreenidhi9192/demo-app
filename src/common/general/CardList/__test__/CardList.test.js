import {
    matchSnapshot,
    snapshotIsMissingSelector,
    snapshotIsMissingSelectorContent
} from '../../../../../__test__/__utils__'

import CardList from '../'

const props = {
    listData: [
        {
            header: 'header 1',
            body: 'body 1'
        },
        {
            header: 'header 2',
            body: 'body 2'
        }
    ]
}

const emptyProps = {
    listData: []
}

const headerlessProps = {
    listData: [
        {
            body: 'body 1'
        },
        {
            body: 'body 2'
        }
    ]
}


describe('CardList:.......................SnapShot Testing:', () => {
    test(
        'CardList Matches jest snapshot',
        matchSnapshot(CardList, props)
    )
    test(
        'Empty CardList Matches jest snapshot',
        matchSnapshot(CardList, emptyProps)
    )
    test(
        'CardList with no contents is missing list items',
        snapshotIsMissingSelector(CardList, emptyProps, '.card:not(.empty-card)')
    )
    test(
        'CardList with cards missing headers have empty headers',
        snapshotIsMissingSelectorContent(CardList, headerlessProps, '.header')
    )
})

