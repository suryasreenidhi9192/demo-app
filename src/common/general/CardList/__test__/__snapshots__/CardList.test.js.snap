// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList:.......................SnapShot Testing: CardList Matches jest snapshot 1`] = `
<ul
  className="card-list"
>
  <li
    className="card"
  >
    <div
      className="header"
    >
      header 1
    </div>
    <div
      className="body"
    >
      body 1
    </div>
  </li>
  <li
    className="card"
  >
    <div
      className="header"
    >
      header 2
    </div>
    <div
      className="body"
    >
      body 2
    </div>
  </li>
  <li
    className="card empty-card"
  />
  <li
    className="card empty-card"
  />
  <li
    className="card empty-card"
  />
</ul>
`;

exports[`CardList:.......................SnapShot Testing: Empty CardList Matches jest snapshot 1`] = `
<ul
  className="card-list"
>
  <li
    className="card empty-card"
  />
  <li
    className="card empty-card"
  />
  <li
    className="card empty-card"
  />
</ul>
`;
